Release PR 1.0.0 (#98)
nf-core/createtaxdb: docs/usage/faq.md

# Frequently asked questions and troubleshooting

## Recommended auxiliary files

Some tools may require or recommend additional files to the reference sequence files - such as taxonomy files - to execute.

We provide a list of required or recommended files, and which pipeline parameters to give them to here:

- bracken
  - taxonomy name dump file (`--namesdmp`)

>   - taxonomy nodes dump file (`--nodesdmp`)

Reviewer: this is an option for all tools. It is slightly confusing to repeat this in the docs this way.

Reviewer: yeah, but only one option for multiple tools.
that's slightly confusing I believe

  - (nucleotide) accession2taxid file (`--accession2taxid`)
- centrifuge
  - taxonomy name dump file (`--namesdmp`)
  - taxonomy nodes dump file (`--nodesdmp`)
  - nucl2taxid file (`--nucl2taxid`)
- diamond
  - taxonomy name dump file (`--namesdmp`)
  - taxonomy nodes dump file (`--nodesdmp`)
  - prot2taxid file (`--prot2taxid`)
- ganon
  - taxonomy name dump file (`--namesdmp`)
  - taxonomy nodes dump file (`--nodesdmp`)
  - genome sizes file (`--genomesizes`)\*

> - kaiju (no additional files required)

Author: ```suggestion
- kaiju (no additional files required)
- kmcp (no additional files required))
```

Reviewer: still need to update this

| 3      | taxid            | The taxon ID corresponding to the names/nodes.dmp                                                                                                                                                               |
| 4      | gi               | The old-style ([now deprecated](https://ncbiinsights.ncbi.nlm.nih.gov/2016/12/06/converting-gi-numbers-to-accession-version/)) NCBI GenInfo Identifier                                                          |

Example:

```tsv title=nucl_db.accession2taxid
accession	accession.version	taxid	gi
NC_018507	CP003708.1	91844	401871806
NZ_CP019811	CP019811.1	1311	1328913993
NZ_CP069563	CP069563.1	817	1986148684
NZ_CP069564	CP069564.1	817	1986152744
NC_012920	J01415.2	9606	113200490
NZ_LS483480	LS483480.1	727	1403648431
MT192765	MT192765.1	694009	1821109001
```

It's important to note that you can have multiple accession numbers within a FASTA file (i.e., one per sequence in the file), therefore accession2taxid mapping files should ensure they reference _all_ accession numbers that are present in the FASTA file.

The accession column should not include the accession version in the first column (i.e. `1`)

### nucl2taxid

> An accession2taxid file is a file that maps database sequence accession IDs (e.g. NCBI GenBank) to their corresponding taxon IDs.

Reviewer: you mention `accession2taxid` but title of section is `nucl2taxid`

Author: It's serves the same purpose, but I've made a suggestion to make it clearer `accession2taxid-like file` (see below))

| Column | Column Name | Description                                                                                                                                                                                                               |
| ------ | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| 1      | accession   | The nucleotide accession ID with version of the sequence - typically embedded as first part of a FASTA header entry (e.g. for `>NC_012920.1 Homo sapiens mitochondrion, complete genome)` the accession is `NC_012829.1`) |
| 2      | taxon id    | The corresponding taxonomy ID of the sequence                                                                                                                                                                             |

Example:

```tsv title="nucl2tax.map"
NC_012920.1	9606
MT192765.1	694009
NZ_CP069563.1	817
NC_018507.1	91844
NZ_CP019811	1311
NZ_LS483480.1	727

```

### prot2taxid

> An accession2taxid file is a file that maps database sequence accession IDs (e.g. NCBI GenBank) to their corresponding taxon IDs.

Author: ```suggestion
An accession2taxid-like file that maps database sequence accession IDs (e.g. NCBI GenBank) to their corresponding taxon IDs.
```
